src/zenml/integrations/vertex/orchestrators/vertex_orchestrator.py

```diff
@@ -0,0 +1,373 @@
+        be executed by the container created using the previously created Docker
+        image.
+
+        This callable is then compiled into a JSON file that is used as the intermediary
+        representation of the Kubeflow pipeline.
+
+        This file then is submitted to the Vertex AI Pipelines service for execution.
+        """
+
+        image_name = self.get_docker_image_name(pipeline.name)
+        image_name = get_image_digest(image_name) or image_name
+
+        def _construct_kfp_pipeline() -> None:
+            """Create a `ContainerOp` for each step which contains the name
+            of the Docker image and configures the entrypoint of the Docker
+            image to run the step.
+
+            Additionally, this gives each `ContainerOp` information about its
+            direct downstream steps.
+
+            If this callable is passed to the `compile()` method of `KFPV2Compiler`
+            all `dsl.ContainerOp` instances will be automatically added to a singular
+            `dsl.Pipeline` instance.
+            """
+            step_name_to_container_op: Dict[str, dsl.ContainerOp] = {}
+
+            for step in sorted_steps:
+                # The command will be needed to eventually call the python step
+                # within the docker container
+                command = VertexEntrypointConfiguration.get_entrypoint_command()
+
+                # The arguments are passed to configure the entrypoint of the
+                # docker container when the step is called.
+                arguments = VertexEntrypointConfiguration.get_entrypoint_arguments(
+                    step=step,
+                    pb2_pipeline=pb2_pipeline,
+                    **{VERTEX_JOB_ID_OPTION: dslv2.PIPELINE_JOB_ID_PLACEHOLDER},
+                )
+
+                # Create the `ContainerOp` for the step. Using the `dsl.ContainerOp`
+                # class directly is deprecated when using the Kubeflow SDK v2.
+                container_op = kfp.components.load_component_from_text(
+                    f"""
+                    name: {step.name}
+                    implementation:
+                        container:
+                            image: {image_name}
+                            command: {command + arguments}"""
+                )()
+
+                # Set upstream tasks as a dependency of the current step
+                upstream_step_names = self.get_upstream_step_names(
+                    step=step, pb2_pipeline=pb2_pipeline
+                )
+                for upstream_step_name in upstream_step_names:
+                    upstream_container_op = step_name_to_container_op[
+                        upstream_step_name
+                    ]
+                    container_op.after(upstream_container_op)
+
+                step_name_to_container_op[step.name] = container_op
+
+        # Save the generated pipeline to a file.
+        assert runtime_configuration.run_name
```

Comment:
Is this just a debugging assertion or should this remain. In case this should stay, it might be good to do this a little more graciously giving an error message in case of an empty run_name

Reply:
To the best of my knowledge, mypy shouldn't complain it if you handle this with an if condition:

```python
if not runtime_configuration.run_name:
    logger.error('Run name is not set ...')
```

Reply:
I put the assertion to follow the style in `KubeflowOrchestrator` https://github.com/zenml-io/zenml/blob/2f9107832455ea2930bf29e28cb3bddc1db0fa7c/src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py#L617

Reply:
fair enough, lets keep it that way then :)